src/fiskaltrust.Launcher/ProcessHost/ProcessHostMonarcStartup.cs

```diff
@@ -31,14 +37,31 @@ public ProcessHostMonarcStartup(ILoggerFactory loggerFactory, ILogger<ProcessHos
             _downloader = downloader;
             _lifetime = lifetime;
             _launcherExecutablePath = launcherExecutablePath;
+            _kestrelReady = new TaskCompletionSource<Uri>();
+
+            hostApplicationLifetime.ApplicationStarted.Register(() =>
+            {
+                try
+                {
+                    _kestrelReady.TrySetResult(new Uri(server.Features.Get<IServerAddressesFeature>()!.Addresses!.First()));
+                }
+                catch (Exception e)
+                {
+                    _kestrelReady.TrySetException(e);
+                }
+            });
         }
 
 
         protected override async Task ExecuteAsync(CancellationToken cancellationToken)
         {
             _lifetime.ApplicationLifetime.ApplicationStopping.Register(() => _logger.LogInformation("Shutting down launcher."));
+            cancellationToken.Register(() => _kestrelReady.TrySetCanceled());
```

Comment:
I guess this is to interupt starting right?

Reply:
Yep, the cancellation token is cancelled when the launcher shuts down and then we don't want to still wait for kestrel to start so we cancel that. I've updated the try catch below so that we don't throw in that case but just shut down like expected.